clar2wasm/tests/wasm-generation/contracts.rs

```diff
@@ -235,4 +235,307 @@ proptest! {
+            write!(call_arguments, "{value} ").unwrap();
+        }
+
+        let second_snippet = format!(
+            r#"
+                (use-trait foo-trait .foo.foo-trait)
+                (define-private (call-it (tt <foo-trait>) {function_arguments})
+                    (let ((ttt tt))
+                        (contract-call? ttt foofun {contract_call_args})
+                    )
+                )
+                (call-it .foo {call_arguments})
+            "#
+        );
+
+        crosscheck_multi_contract(
+            &[
+                (first_contract_name, &first_snippet),
+                (second_contract_name, &second_snippet),
+            ],
+            Ok(Some(Value::Response(ResponseData {
+                committed: true,
+                data: Box::new(result.into()),
+            }))),
+        );
+    }
+
+    #[test]
+    fn contract_dynamic_call_accepts_any_args_trait_in_match_some(
+        (tys, values)
+            in prop::collection::vec(
+                prop_signature().prop_ind_flat_map2(|ty| PropValue::from_type(ty.clone())),
+                1..=20
+            )
+            .prop_map(|arg_ty| arg_ty.into_iter().unzip::<_, _, Vec<_>, Vec<_>>())
+            .no_shrink(),
+        result in PropValue::any().no_shrink(),
+        err_type in prop_signature().no_shrink(),
+    ) {
+        // first contract
+        let first_contract_name = "foo".into();
+        let mut function_types = String::new();
+        let mut function_arguments = String::new();
+        for (name, ty) in ('a'..).zip(tys.iter()) {
+            let ty = type_string(ty);
+            write!(function_arguments, "({name} {ty}) ").unwrap();
+            function_types += &(ty + " ");
+        }
+        let first_snippet = format!(
+            r#"
+                (define-trait foo-trait (
+                    (foofun ({function_types}) (response {} {}))
+                ))
+
+                (define-public (foofun {function_arguments})
+                    (ok {result})
+                )
+            "#,
+            result.type_string(),
+            type_string(&err_type)
+        );
+
+        // second contract
+        let second_contract_name = "bar".into();
+
+        let contract_call_args: String =
+            ('a'..)
+                .take(values.len())
+                .fold(String::new(), |mut acc, name| {
+                    write!(acc, "{name} ").unwrap();
+                    acc
+                });
+
+        let mut call_arguments = String::new();
+        for value in values {
+            write!(call_arguments, "{value} ").unwrap();
+        }
+
+        let second_snippet = format!(
+            r#"
+                (use-trait foo-trait .foo.foo-trait)
+                (define-private (call-it (tt (optional <foo-trait>)) {function_arguments})
+                    (match tt
+                        ttt (contract-call? ttt foofun {contract_call_args})
+                        (ok {result})
```

Comment:
nit: return something different from `(ok {result})` in the none arm to differentiate if from the contract call? also in the match response tests below

Reply:
I wanted some kind of `panic`, but this doesn't exist in Clarity it seems :(

Reply:
Done in [a2c4a86](https://github.com/stacks-network/clarity-wasm/pull/483/commits/a2c4a868c436f6640fb7bee33df1b34f39738c89)